packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -345,8 +663,189 @@ markdown based on [CommonMark](http://commonmark.org/) flavor:
+specifically tailored for user mentions. These values are kept as part of the `RichTextMentionInternalsOptions` and will be utilized by various components that require access to these specific values.
+
+Validation will be integrated into the internal workings through a `validConfiguration` flag. There is a base class called `RichTextMentionValidator`
+responsible for handling validation, with methods for setting and retrieving the validation configuration. This approach ensures that the state of
+valid and invalid values passed within the mapping element is effectively managed. The class obtains the name of the validity flag to communicate this
+information via a public API called `validity`.
+
+By deriving from the base, the mention options can validate the following conditions for the `key` and `text` value in mapping element:
+
+1. `validateKeyValuesForType(keys, keyType)`
+2. `validateMappingTypes(mappings)`
+3. `validateUniqueKeys(keys, keyType)`
+4. `validateNoMissingKeys(mappings)`
+5. `validateNoMissingText(mappings)`
+
+_Note_: These are subject to change based on the property changes in the `nimble-mapping-mention-user` element.
+
+If any of the mention option is invalid as per the above validation, that particular option will render as an empty
+option in the list. This indicates the client that some of the option is wrongly configured and identify the
+validation details using the public API `validity`.
+
+The public API to determine the validity of the mention options are `checkValidity()` and `validity`.
+See the [API section](#api) for more details.
+
+#### 5. _nimble-rich-text-user-mention-view_:
+
+The foundation for configuring the rendering of mention nodes in the UI is provided by a base class known as `nimble-rich-text-mention-view`. This base class is an extension of
+`FoundationElement`, and its template contains a `span` element with the required CSS styling. It is equipped with attributes such as `data-id`, `data-label`, and `data-type`,
+along with properties like `character`, which is included in the display alongside mentioned users.
+
+The `nimble-rich-text-user-mention-view` is a view class that is derived from the above base class and makes use of the template defined in the base class.
+It customizes the character property of the base class by setting it to **"@"** in order to render the mention node with the **"@"** symbol as a prefix in the text content of the node.
+
+The Mention node from Tiptap is extended, and the `renderHTML` method is overridden to render the element as a `nimble-rich-text-user-mention-view` instead of the default `span`
+element as shown in the [Tiptap code](https://github.com/ueberdosis/tiptap/blob/42039c05f0894a2730a7b8f1b943ddb22d52a824/packages/extension-mention/src/mention.ts#L112).
+The attributes will be retained and included in the rendered mention nodes.
```

Comment:
Just a note that I noticed some issues in the prototype when creating an @mention while the style was set to italics. The new @mention would start out with no styling. Conversely if the styling was bold then the new @mention would successfully be bolded. I could also successfully select an existing @mention and change it to be italics.

This doesn't need to be part of the spec, just some test cases for your development.

Reply:
We have recently aligned with the interaction designer who created those mockups for `@mention` and mentioned that all the mentioned node's default values should be bold and users have the option to change the formatting whenever they need. I linking here the [discussion on Teams (Internal link)](https://teams.microsoft.com/l/message/19:a045e5de02a8458484bd3705ac3df6ed@thread.skype/1697222489787?tenantId=87ba1f9a-44cd-43a6-b008-6fdb45a5204e&groupId=41626d4a-3f1f-49e2-abdc-f590be4a329d&parentMessageId=1697174622121&teamName=ASW%20SystemLink&channelName=LIMS%20UI-UX&createdTime=1697222489787) for more details.

I have included this in the `States` section for `@mention` in the doc.